### PR TITLE
[BugFix] Fix generated column resolved fail in subquery

### LINKED ISSUE #33252: When the table contains generated column fields, CTE syntax error is reported
create table test_gen_column (
    evt_date date NOT NULL comment '统计日期',
    evt_week DATE AS date_trunc('week', evt_date) comment '事件-所属周第一天',
    evt_month DATE AS date_trunc('month', evt_date) comment '事件-所属月第一天'
)
ENGINE=olap
PRIMARY KEY(evt_date)
DISTRIBUTED BY HASH(evt_date)
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false");
;

insert into test_gen_column(evt_date) values ('2023-10-20');

select evt_date, evt_week,evt_month from test_gen_column;

with test_cte as (
    select * from test_gen_column
)
select * from test_cte;

The above statement will report the following error：

[42000][1064] Getting analyzing error. Detail message: Column '`test`.`test_gen_column`.`evt_week`' cannot be resolved.
---------------

### StarRocks version (Required)
 3.1.2-4f3a2ee

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #33252  #32931

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by srlch — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|7|7|100.00%|[]|

**Review by liuyehcf — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java [hunk 1]
```diff
@@ -199,15 +198,20 @@ public Scope visitSelect(SelectRelation selectRelation, Scope scope) {
...
-                        generatedExprToColumnRef.putAll(tableRelation.getGeneratedExprToColumnRef());
-                        return null;
-                    }
-                }.visit(resolvedRelation);
-            }
+            new AstVisitor<Void, Void>() {
+                @Override
+                public Void visitTable(TableRelation tableRelation, Void context) {
+                    generatedExprToColumnRef.putAll(tableRelation.getGeneratedExprToColumnRef());
+                    return null;
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:205]: Do we need process other relation type?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java:205]: No, currently only direct rewriting of generated columns is supported.

**Review by packy92 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**